core/capabilities/ccip/configs/evm/contract_reader.go

```diff
@@ -25,6 +27,17 @@ import (
 	evmrelaytypes "github.com/smartcontractkit/chainlink/v2/core/services/relay/evm/types"
 )
 
+const (
+	// CommitExecLogsRetention defines the duration for which logs critical for Commit/Exec plugins processing are retained.
+	// Although Exec relies on permissionlessExecThreshold which is lower than 24hours for picking eligible CommitRoots,
+	// Commit still can reach to older logs because it filters them by sequence numbers. For instance, in case of RMN curse on chain,
+	// we might have logs waiting in OnRamp to be committed first. When outage takes days we still would
+	// be able to bring back processing without replaying any logs from chain. You can read that param as
+	// "how long CCIP can be down and still be able to process all the messages after getting back to life".
+	// Breaching this threshold would require replaying chain using LogPoller from the beginning of the outage.
+	CommitExecLogsRetention = 30 * 24 * time.Hour // 30 days
```

Comment:
I would rename it to `DefaultCCIPLogsRetentions` or sth like this, because you applied it to all contracts. It was named like these in 1.5 because we had more than retention

Reply:
Please add links to the origin if it's taken directly from 1.5

[https://github.com/smartcontractkit/ccip/blob/80eb41b42d0a9807d6949fa0f5ee6c40c8b7[…]7d1/core/services/ocr2/plugins/ccip/internal/ccipdata/reader.go](https://github.com/smartcontractkit/ccip/blob/80eb41b42d0a9807d6949fa0f5ee6c40c8b727d1/core/services/ocr2/plugins/ccip/internal/ccipdata/reader.go#L24)
https://github.com/smartcontractkit/ccip/pull/530